### ISSUE TITLE
Installing gmt==6.0.0 on Pangeo

### ISSUE THREAD

**Issue by tjcrone** [pangeo-data/pangeo-cloud-federation#525]
I am trying to conda install gmt==6.0.0 on ooi.pangeo.io, which runs on an image specified (repo2docker) here: https://github.com/pangeo-data/pangeo-cloud-federation/tree/staging/deployments/ooi/image/binder

Another way to get the OOI image is:
```
docker pull ooicloud.azurecr.io/ooi-pangeo-io-notebook:c987384
```
A conda install of gmt==6.0.0 takes a *long* time to solve, and fails with uncountable conflicts. It's pretty hard to parse the error which is very long. @ocefpaf, any thoughts on how to troubleshoot this?

cc @seisman, @leouieda

**Comment by ocefpaf** [comment]
Debugging that in a stacked install like pangeo may take a while. The env here is OK with `gmt` and you can test it with:

```
conda create --name TEST av erddapy pip gmt
```

However, this env has already many packages installed from [`pangeo-notebook`](https://github.com/pangeo-data/pangeo-cloud-federation/blob/staging/deployments/ooi/image/binder/Dockerfile#L2), right? I'm checking those now to see if I can find the conflict. 

---
Edit: I'm using [this one](https://github.com/pangeo-data/pangeo-stacks/blob/master/pangeo-notebook/binder/environment.yml), let me know if that is the correct one for this test.

**Comment by tjcrone** [comment]
Right. Lots of packages from the base Pangeo notebook. You are right that we can do our work inside of a conda env, but we had some environment configuration conflicts that were [looking hard to solve](https://github.com/GenericMappingTools/pygmt/issues/393), so I thought that an install of gmt=6.0.0 in the base environment might be a better solution. Unclear at this stage which option is easier. The far better option is to figure out how to install gmt==6.0.0 in the base notebook image. Thank you for any help you can provide!

**Comment by tjcrone** [comment]
@ocefpaf, regarding your question, I think you are on the correct environment.yaml file. But remember this is also [repo2docker](https://github.com/jupyter/repo2docker), so the other files in that repo directory come into play.

**Comment by ocefpaf** [comment]
> You are right that we can do our work inside of a conda env,

Oh no. I'm not suggesting you do that (even though it is recommended), I'm just pointing that this is how we debug this. I'm using now a modified env file that has https://github.com/pangeo-data/pangeo-stacks/blob/master/pangeo-notebook/binder/environment.yml and https://github.com/pangeo-data/pangeo-cloud-federation/blob/staging/deployments/ooi/image/binder/environment.yml in a single env. I got conflicts! Let the debugging begin!!

**Comment by ocefpaf** [comment]
> But remember this is also repo2docker, so the other files in that repo directory come into play.

Yep. I know but that is virtually impossible to debug due to their decision to rely on the broken channel and not activate `strict` channel option. I can debug up to the first level and using those config options, after that there is very little I can do.

**Comment by tjcrone** [comment]
Okay. Well, thank you very much for any help you can provide. I very much appreciate it. Regarding the stack, we should be careful that we are following the right stack because it's a bit confusing.

How would you feel about looking at the conda install gmt==6.0.0 running inside our Docker image? Is that a waste of time since the solve comes after the fact?

**Comment by ocefpaf** [comment]
> How would you feel about looking at the conda install gmt==6.0.0 running inside our Docker image? Is that a waste of time since the solve comes after the fact?

Kind of both. First I need some trial-and-error to find any obvious conflicts in the `environment.yaml` (still doing that, it takes a while). If I cannot find any problem there then I need to use the docker image so I can try to find what config option is different that may be the source of the problem. For this case, b/c `gmt` pulls `gdal`, I'm 80% confident I'll find a conflict and either fix it or, at least, have a package to blame.

**Comment by ocefpaf** [comment]
OK, the offending packages are `tensorflow`, `pytorch-cpu`, `keras`, and `basemap`. All but `basemap` are probably "fixable" as we re-build and add them in `conda-forge` b/c the conflicts are due to different builds from different channels. `basemap` on the other hand would be quite hard b/c it is abandoned and it does not work with latest `proj 6` that we are using on `conda-forge`.

I'm working on the others and I'll report back here if I can make them play together with `gmt=6`.

---
edit: using `python=3.6` instead of forcing latest `python=3.8` I can install `keras`, `tensorflow` and `pytorch-cpu`, leaving only `basemap` out. Looks like you'all do not enforce a `python` version. So, if `pangeo` is willing to drop `basemap`, you should be good to go there. (Installing those in python=3.8 should be able as soon as we rebuild them in `conda-forge`.)

**Comment by tjcrone** [comment]
Filipe you are amazing.

**Comment by tjcrone** [comment]
Thank you very much for taking a look at this Filipe! I will check with @jhamman and others to see if perhaps it is time to drop basemap since it has been abandoned. Might be worth doing. Cheers!

**Comment by leouieda** [comment]
@ocefpaf thanks for all the debugging work here. I would not have been able to help since I had no idea where to begin. Bookmarking this for later :)

**Comment by tjcrone** [comment]
Closing because I was able to install gmt=6 by refactoring the OOI image to start with base-notebook rather than pangeo-notebook, removing the naughty naughty packages, and building from there. Thanks everyone for your help!
